components/omega/src/infra/MetaData.cpp

```diff
@@ -0,0 +1,435 @@
+std::shared_ptr<MetaData> MetaData::create(const std::string Name) {
+
+   if (has(Name)) {
+      throw std::runtime_error("Failed to create a field instance because '" +
+                               Name + "' already exists.");
+   }
+
+   auto Data = std::make_shared<MetaData>();
+
+   AllFields[Name] = Data;
+
+   return Data;
+}
+
+std::shared_ptr<MetaData> MetaData::create(
+    const std::string Name,
+    const std::initializer_list<std::pair<std::string, std::any>> &MetaPairs) {
+
+   auto Data = create(Name);
+
+   for (const auto &MetaPair : MetaPairs) {
+      Data->addMetaData(MetaPair.first, MetaPair.second);
+   }
+
+   return Data;
+}
+
+/// Create metadata for a variable/field using all
+/// required metadata. Note that if input parameters don't exist
+/// (eg stdName) or don't make sense (eg min/max or fill) for a
+/// given field, empty or 0 entries can be provided. Similarly
+/// for scalars, nDims can be 0 and an emtpy MetaDim vector can be
+/// supplied.
+
+std::shared_ptr<MetaData> ArrayMetaData::create(
+    const std::string Name,        /// Name of var
+    const std::string Description, /// long Name or description
+    const std::string Units,       /// units
+    const std::string StdName,     /// CF standard Name
+    const std::any ValidMin,       /// min valid field Value
+    const std::any ValidMax,       /// max valid field Value
+    const std::any FillValue,      /// scalar used for undefined entries
+    const int NDims,               /// number of dimensions
+    const std::vector<std::shared_ptr<MetaDim>> Dimensions // dim pointers
+) {
+
+   auto Data = MetaData::create(Name);
+
+   Data->addMetaData("Description", Description);
+   Data->addMetaData("Units", Units);
+   Data->addMetaData("StdName", StdName);
+   Data->addMetaData("ValidMin", ValidMin);
+   Data->addMetaData("ValidMax", ValidMax);
+   Data->addMetaData("FillValue", FillValue);
+   Data->addMetaData("NDims", NDims);
+   Data->addMetaData("Dimensions", Dimensions);
+
+   return Data;
+}
+
+int MetaData::destroy(const std::string Name /// Name of field to remove
+) {
+   int RetVal = 0;
+
+   if (!has(Name)) {
+      LOG_ERROR("Failed to destroy the field '" + Name + "' because " +
+                "the field name does not exist.");
+      RetVal = -1;
+
+   } else {
+      if (AllFields.erase(Name) != 1) {
+         LOG_ERROR("Field, '" + Name + "', is not correctly removed.");
+         RetVal = -2;
+      }
+   }
+
+   return RetVal;
+}
+
+std::shared_ptr<MetaData> MetaData::get(const std::string Name /// name of field
+) {
+   if (!has(Name)) {
+      throw std::runtime_error("Failed to retrieve a field instance because '" +
+                               Name + "' does not exist.");
+   }
+
+   return AllFields[Name];
+}
+
+bool MetaData::hasMetaData(const std::string Name // name of metadata
+) {
+   return (MetaMap.find(Name) != MetaMap.end());
+}
+int MetaData::addMetaData(
```

Comment:
```suggestion
}

int MetaData::addMetaData(
```

Reply:
Thanks for the comment. I will apply this in next commit.